File: DotNet/ModelTracKer/Models/Tracker.cs
using Microsoft.Identity.Client;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ModelTracKer.Models
{
    [Table("tracker")]
    public class Tracker
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int Tracker_id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Tracker_Name { get; set; }

        [Required]
        [MaxLength(200)]
        public string Client_Name { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public double Investment { get; set; }

        [Required]
        public bool GenAiAdoptation {  get; set; }


        public int GenAiTool_Id { get; set; }

        [ForeignKey("GenAiTool_Id")]
        public GenAiTool? GenAiTool { get; set; }


        public int ReasonForNoGenAiAdoptation_Id { get; set; }

        [ForeignKey("ReasonForNoGenAiAdoptation_Id")]
        public ReasonForNoGenAiAdoptation? ReasonForNoGenAiAdoptation { get; set; }


        public ICollection<opp_Accelerator> oppAccelerators { get; set; }

        public ICollection<opp_microservice> oppMicroservices { get; set; }

        public ICollection<opp_competition> oppCompetitions { get; set; }
    }
}

File: Java/tracker/tracker/src/main/java/com/tacker/tracker/models/Tracker.java
package com.tacker.tracker.models;

import jakarta.persistence.*;
import lombok.*;


import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "tracker")
//@Data // Generates getters, setters, toString, equals, hashCode
@Getter
@Setter
@NoArgsConstructor // Generates no-args constructor required by JPA
@AllArgsConstructor // Generates all-args constructor
@ToString
public class Tracker {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "tracker_id", nullable = false)
    private int trackerId;

    @Column(name = "tracker_name", length = 500, nullable = false)
    private String trackerName;

    @Column(name = "client_name", length = 500, nullable = false)
    private String clientName;

    @Column(nullable = false)
    private double investment;

    @Column(name = "gen_ai_adoptation", nullable = false)
    private boolean genAiAdoptation;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "gen_ai_tool_id")
    private GenAiTool genAiTool;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "reason_for_no_gen_ai_adoptation_id")
    private ReasonForNoGenAiAdoptation reasonForNoGenAiAdoptation;

    @OneToMany(mappedBy = "tracker", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    private Set<OppAccelerator> oppAccelerators = new HashSet<>();

    @OneToMany(mappedBy = "tracker", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    private Set<OppMicroservice> oppMicroservices = new HashSet<>();

    @OneToMany(mappedBy = "tracker", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    private Set<OppCompetition> oppCompetitions = new HashSet<>();

}